app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -331,6 +335,24 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
         return $order;
     }
 
+    /**
+     * Associate order to customer when placing on product detail page
+     * @param $orderIncrementId
+     */
+    protected function associateOrderToCustomerWhenPlacingOnPDP($orderIncrementId){
+        $customer = Mage::getSingleton('customer/session')->getCustomer();
+
+        $order = Mage::getModel('sales/order')->loadByIncrementId($orderIncrementId);
+        $order->setCustomerId($customer->getId())
+            ->setCustomerEmail($customer->getEmail())
+            ->setCustomerFirstname($customer->getFirstname())
```

Comment:
not sure if we want to override name. looks like we don't do this for standard bolt checkout?

Reply:
> is it possible to add test?

@daisy1754 I wouldn't recommend it here as the changes are for saving data, and it doesn't really test functionality as much as it tests reading and writing data to the database. Let me know your thoughts on this.

Also, can you explain what you mean by "not sure if we want to override name. looks like we don't do this for standard bolt checkout?"

Reply:
I was looking at order creation logic for standard bolt checkout - https://github.com/BoltApp/bolt-magento1/blob/f073380d2b8a1b6292a0ceb4155157632830f35b/app/code/community/Bolt/Boltpay/Model/Order.php#L104

Looks like we set name for guest checkout, but we don't set name for logged in checkout (I assume we prefer logged in user name in magento, as opposed to name entered in our checkout?). Not strong opinion here but I think it'd be good if we're consistent.

Reply:
Yes, for guest checkout, it will use the name that the customer uses in the Bolt checkout modal and for logged in customer, it uses the name from the logged in customer (not the one that they input in the Bolt modal). I tested to confirm and the changes I made function the same way that regular Bolt checkout does.